### PR TITLE
Fix warning when running scripts/build_ios.sh

### PR DESCRIPTION
**Pull request by skyline75489**
* Fixes `cmake implicitly converting 'string' to 'STRING' type`
* Fixes `clang: warning: argument unused during compilation: '-mfpu=neon-fp16' [-Wunused-command-line-argument]`

### CODE REVIEW

File: cmake/MiscCheck.cmake [hunk 1]
```diff
@@ -306,8 +306,9 @@ endif()
 # Also, we will turn off deprecated-declarations
 # due to protobuf.
 
-if(IOS)
+if(IOS AND ${IOS_ARCH} MATCHES "armv7*")
```
Comment (skyline75489) [inline cmake/MiscCheck.cmake:309]: Seems that `-mfpu` is only recognized when targeting `armv7` & `armv7k`.

File: cmake/MiscCheck.cmake [hunk 2]
```diff
@@ -306,8 +306,9 @@ endif()
 # Also, we will turn off deprecated-declarations
 # due to protobuf.
 
-if(IOS)
+if(IOS AND (${IOS_ARCH} MATCHES "armv7*"))
   add_definitions("-mfpu=neon-fp16")
+  add_definitions("-arch" ${IOS_ARCH})
```
Comment (skyline75489) [inline cmake/MiscCheck.cmake:311]: I found that XNNPACK has more comprehensive flag configuration https://github.com/google/XNNPACK/blob/02bb42942798fdd0cf444b0fc90b6e20b847688f/CMakeLists.txt#L2729. Not sure if we want to use this, too.

Comment (skyline75489) [reply cmake/MiscCheck.cmake:311]: To explain this change. `-mfpu=neon-fp16` is only supported on armv7 devices (iPhone 4, 4s, iPhone 5, 5c, 5s). Nowadays we would normally target arm64 which `-march=armv8.2-a+fp16` and related should be used.

File: cmake/iOS.cmake [hunk 3]
```diff
@@ -42,7 +42,7 @@ set(CMAKE_OSX_DEPLOYMENT_TARGET "" CACHE STRING "Force unset of the deployment t
 find_program(CMAKE_UNAME uname /bin /usr/bin /usr/local/bin)
 if(CMAKE_UNAME)
     exec_program(uname ARGS -r OUTPUT_VARIABLE CMAKE_HOST_SYSTEM_VERSION)
-    string(REGEX REPLACE "^([0-9]+)\\.([0-9]+).*$" "\\1" DARWIN_MAJOR_VERSION "${CMAKE_HOST_SYSTEM_VERSION}")
+    STRING(REGEX REPLACE "^([0-9]+)\\.([0-9]+).*$" "\\1" DARWIN_MAJOR_VERSION "${CMAKE_HOST_SYSTEM_VERSION}")
```
Comment (malfet) [inline cmake/iOS.cmake:45]: Please keep the command lowercase while constants uppercase
```suggestion
    string(REGEX REPLACE "^([0-9]+)\\.([0-9]+).*$" "\\1" DARWIN_MAJOR_VERSION "${CMAKE_HOST_SYSTEM_VERSION}")
```

**Review by malfet — APPROVED** [review]

**Review by malfet — APPROVED** [review]

**Pull request closed without merging**